src/main/java/net/dv8tion/jda/api/JDA.java

```diff
@@ -551,17 +555,23 @@ default JDA awaitReady() throws InterruptedException
      * @throws IllegalArgumentException
      *         If null is provided
      *
-     * @return {@link CommandCreateAction}
+     * @return {@link RestAction} - Type: {@link Command}
+     *         <br>The RestAction used to create or update the command
      *
-     * @see net.dv8tion.jda.api.entities.Guild#upsertCommand(CommandData)
+     * @see    Commands#slash(String, String) Commands.slash(...)
+     * @see    Commands#message(String) Commands.message(...)
+     * @see    Commands#user(String) Commands.user(...)
+     * @see    Guild#upsertCommand(CommandData) Guild.upsertCommand(...)
      */
     @Nonnull
     @CheckReturnValue
-    CommandCreateAction upsertCommand(@Nonnull CommandData command);
+    RestAction<Command> upsertCommand(@Nonnull CommandData command);
```

Comment:
To prevent code which adds options to context menu commands. Context menus can only really have a name and thats it. I presumed with this that everything is already set when you use the CommandData parameter.